src/Field/BasicField.php

```diff
@@ -115,10 +124,11 @@ public function fetchFieldKey($fieldName)
      */
     public function fetchFieldType($fieldKey)
     {
-        $post = $this->post->orWhere(function ($query) use ($fieldKey) {
-            $query->where('post_name', $fieldKey);
-            $query->where('post_type', 'acf-field');
-        })->first();
+        $post = Post::on($this->post->getConnectionName())
```

Comment:
@jgrossi This on method is already used in previous PR's, this sets the correct database connection. What looks strange for you?